grpc/creds/creds.go

```diff
@@ -73,7 +73,10 @@ func (tc *clientTransportCredentials) ClientHandshake(ctx context.Context, addr
 	case err := <-errChan:
 		if err != nil {
 			_ = rawConn.Close()
-			return nil, nil, fmt.Errorf("boulder/grpc/creds: TLS handshake failed: %s", err)
+			// directly return the error so that gRPC can use its
```

Comment:
Let's rewrite this comment slightly:

// IMPORTANT: Don't wrap this error. gRPC expects to be able to check err.Temporary to spot temporary errors and reconnect when they happen.

Also, above in the ctx.Done() case, should we avoid wrapping that?